src/main/java/org/jenkinsci/plugins/workflow/job/WorkflowRun.java

```diff
@@ -603,6 +417,13 @@ private void finish(@Nonnull Result r, @CheckForNull Throwable t) {
                 completed.set(true);
             }
         }
+        Executor executor = getExecutor();
+        if (executor != null) {
+            AsynchronousExecution asynchronousExecution = executor.getAsynchronousExecution();
+            if (asynchronousExecution != null) {
+                asynchronousExecution.completed(null);
```

Comment:
Uses to be part of `copyLogsTask` (line 278); that is now gone, so we need to somehow release the flyweight executor slot. This seems like the appropriate place to do it.